src/RuleList.js

```diff
@@ -155,7 +155,9 @@ export default class RuleList {
       const cssRule = cssRules[i]
       let key = this.options.sheet.renderer.getKey(cssRule)
       if (map[key]) key = map[key]
-      const rule = this.map[key]
+
+      const mappedGlobal: any = this.map['@global']
```

Comment:
core doesn't know anything about plugins, it needs to be solved locally by a plugin, if plugin is missing what it needs, core should provide a generic solution so that plugin can fix it

Reply:
@kof Thats what I was thinking that this should be moved to the plugin itself. On this weekend if I get some time I will try to move it there. as of right now I am not sure how the plugin should handle this.

Reply:
hey @sambhav-gore I see you got distracted, any chance you can finish this one?